From da9e08ad23f3d19c8d30fd49aaf3991be736ddc2 Mon Sep 17 00:00:00 2001
From: Lukas Bezdicka <lbezdick@redhat.com>
Date: Wed, 17 Oct 2018 00:10:26 +0200
Subject: [PATCH] Migrate IPs off the Pacemaker node before update

During testing I found APIs become unavailable because of pcs
cluster stop command. This is caused by wrong kind of order
constraint. Sadly fix would be delivered by puppet running after
the fact so we move the IPs manually. Also in order to be less
racy we always should use --wait.

Change-Id: I939464b93df2c33aa662a1598c8b4f478b798ec2
---
 extraconfig/tasks/yum_update.sh | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/extraconfig/tasks/yum_update.sh b/extraconfig/tasks/yum_update.sh
index 66d0242f7..6928c9de6 100755
--- a/extraconfig/tasks/yum_update.sh
+++ b/extraconfig/tasks/yum_update.sh
@@ -105,7 +105,14 @@ if [[ "$pacemaker_status" == "active" ]] ; then
         echo "Active node count is 1, stopping node with --force"
         pcs cluster stop --force
     else
-        pcs cluster stop
+        hostname=$(hostname -s)
+       resources=$(pcs resource show | grep IP | grep $hostname | awk '{print $1}')
+       for resource in ${resources};
+       do
+            pcs resource move $resource --wait=300
+           pcs constraint location remove cli-ban-$resource-on-$hostname  --wait=300
+        done
+        pcs cluster stop --wait=300
     fi
     update_network
 else
@@ -131,7 +138,7 @@ echo "yum return code: $return_code"

 if [[ "$pacemaker_status" == "active" ]] ; then
     echo "Starting cluster node"
-    pcs cluster start
+    pcs cluster start --wait=300

     hostname=$(hostname -s)
     tstart=$(date +%s)
-- 
2.17.2

